src/probnum/linops/_scaling.py

```diff
@@ -124,6 +124,8 @@ def __init__(
 
             trace = lambda: self.shape[0] * self._scalar
         elif np.ndim(factors) == 1:
+            if np.all(factors == factors[0]):
+                self._scalar = factors[0]
```

Comment:
Personally, I would not do this check here, but rather implement a cached property which computes it on demand.
This avoids potentially unnecessary constructor overhead